src/Model.php

```diff
@@ -826,18 +1057,30 @@ public function getIterator()
         $this->unload();
     }
 
+    /**
+     * Returns iterator.
+     *
+     * @return Iterator
+     */
     public function rawIterator()
     {
         return $this->persistence->prepareIterator($this);
     }
 
+    /**
+     * Executes specified method or callback for each record in DataSet.
+     *
+     * @param string|callable $method
+     *
+     * @return $this
+     */
     public function each($method)
     {
         foreach ($this as $rec) {
             if (is_string($method)) {
                 $rec->$method();
-            } else {
-                $method($rec);
+            } elseif (is_callable($method)) {
```

Comment:
warning, this will fail when 'system' variable is passed, because it will think it is executable. Search for 'Callable' for a proper implementation.